vars/neoDeploy.groovy

```diff
@@ -8,222 +11,218 @@ import com.sap.piper.tools.ToolDescriptor
+                if (!isAppRunning(neoCommandHelper)) {
+                    warAction = 'deploy'
+                    echo "Rolling update not possible because application is not running. Falling back to standard deployment."
+                }
             }
-        }
 
-        if (deployMode == 'warParams') {
-            applicationName = utils.getMandatoryParameter(configuration, 'applicationName')
-            runtime = utils.getMandatoryParameter(configuration, 'runtime')
-            runtimeVersion = utils.getMandatoryParameter(configuration, 'runtimeVersion')
-            def vmSizes = ['lite', 'pro', 'prem', 'prem-plus']
-            vmSize = configuration.vmSize
-            if (! (vmSize in vmSizes)) {
-                throw new Exception("[neoDeploy] Invalid vmSize = '${vmSize}'. Valid 'vmSize' values are: ${vmSizes}.")
+            echo "Link to the application dashboard: ${neoCommandHelper.cloudCockpitLink()}"
+
+            if (warAction == 'rolling-update') {
+                sh neoCommandHelper.rollingUpdateCommand()
+            } else {
+                sh neoCommandHelper.deployCommand()
+                sh neoCommandHelper.restartCommand()
             }
-        }
 
-        if (deployMode in ['mta','warParams']) {
-            deployHost = utils.getMandatoryParameter(configuration, 'host')
-            deployAccount = utils.getMandatoryParameter(configuration, 'account')
-        }
 
-        def neo = new ToolDescriptor('SAP Cloud Platform Console Client', 'NEO_HOME', 'neoHome', '/tools/', 'neo.sh', null, 'version')
-        def neoExecutable = neo.getToolExecutable(this, configuration)
-        def neoDeployScript = """#!/bin/bash
-                                 "${neoExecutable}" ${warAction} \
-                                 --source "${archivePath}" \
-                              """
-
-        if (deployMode in ['mta', 'warParams']) {
-            neoDeployScript +=
-                    """--host '${deployHost}' \
-                    --account '${deployAccount}' \
-                    """
-        }
+        } else if (deployMode == 'mta') {
+            warAction = 'deploy-mta'
 
-        if (deployMode == 'mta') {
-            neoDeployScript += "--synchronous"
+            sh neoCommandHelper.deployMta()
         }
+    }
+    catch (Exception ex) {
+        echo "Error while deploying to SAP Cloud Platform. Here are the neo.sh logs:"
+        sh "cat ${neoToolDescriptor.getToolLocation(script, configuration)}/tools/log/*"
+        throw ex
+    }
+}
 
-        if (deployMode == 'warParams') {
-            neoDeployScript +=
-                    """--application '${applicationName}' \
-                    --runtime '${runtime}' \
-                    --runtime-version '${runtimeVersion}' \
-                    --size '${vmSize}'"""
-        }
+private boolean isAppRunning(NeoCommandHelper commandHelper) {
+    def status = sh script: "${commandHelper.statusCommand()} || true", returnStdout: true
```

Comment:
[question] is there a possibility to get the answer via the return value of the neo toolset rather than parsing the command line output which is IMO a second best approach?

Reply:
Unfortunately not.  In both cases (application is stopped and application is running) the exit code is 0. In case that the application does not exist, the exit code is 1 (general error).